server/models/questions/addEditor.js

```diff
@@ -12,11 +12,16 @@ const { pool } = require('../database')
 function addEditor({ username, editor, questionId }) {
   return new Promise((resolve, reject) => {
     pool.query(
-      `INSERT INTO questions_editors(question_id, editor) 
-       SELECT question_id, ? FROM questions_editors WHERE editor=? AND question_id=?`,
-      [editor, username, questionId],
+      `INSERT INTO questions_editors(question_id, editor, access) 
+       SELECT question_id, ?, ? FROM questions_editors WHERE editor=? AND question_id=? AND access=?
+       ON DUPLICATE KEY UPDATE access=?`,
+      [editor, 'write', username, questionId, 'write', 'write'],
       (error, results) => {
         if (error || !results) {
+          const { code } = error
+          if (code === 'ER_DUP_ENTRY') {
+            return reject('The user already has write access to the question')
+          }
```

Comment:
This is redundant as it will not error out if there is duplicate entry, rather it will just update `access` to `write`
```suggestion
```